modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java

```diff
@@ -664,6 +665,101 @@ public void testAddObjectEntryAsDraft() throws Exception {
 		}
 	}
 
+	@Test
+	public void testAddObjectEntryWithAttachmentObjectField() throws Exception {
+		FileEntry tempFileEntry1 = _addTempFileEntry(StringUtil.randomString());
+
+		ObjectEntry objectEntry = _addObjectEntry(
+			HashMapBuilder.<String, Serializable>put(
+				"emailAddressRequired", "james@liferay.com"
+			).put(
+				"listTypeEntryKeyRequired", "listTypeEntryKey1"
+			).put(
+				"upload", tempFileEntry1.getFileEntryId()
+			).build());
+
+		long tempFileEntryId = tempFileEntry1.getFileEntryId();
+
+		AssertUtils.assertFailure(
+			NoSuchFileEntryException.class,
+			StringBundler.concat(
+				"No FileEntry exists with the key {fileEntryId=",
+				tempFileEntryId, "}"),
+			() -> _dlAppLocalService.getFileEntry(tempFileEntryId));
+
+		Map<String, Serializable> values = objectEntry.getValues();
+
+		Assert.assertNotEquals(tempFileEntryId, values.get("upload"));
+
+		DLFileEntry persistedFileEntry = _dlFileEntryLocalService.getFileEntry(
+			(Long)values.get("upload"));
+
+		Assert.assertEquals(
+			_objectDefinition.getClassName(),
+			persistedFileEntry.getClassName());
+		Assert.assertEquals(
+			objectEntry.getObjectEntryId(), persistedFileEntry.getClassPK());
+
+		// LPS-180587 Partial updates should not delete existing files
+
+		_objectEntryLocalService.updateObjectEntry(
+			TestPropsValues.getUserId(), objectEntry.getObjectEntryId(),
+			HashMapBuilder.<String, Serializable>put(
+				"state", "listTypeEntryKey3"
+			).build(),
+			ServiceContextTestUtil.getServiceContext());
+
+		long persistedFileEntryId1 = persistedFileEntry.getFileEntryId();
+
+		Assert.assertNotNull(
+			_dlAppLocalService.getFileEntry(persistedFileEntryId1));
+
+		_objectEntryLocalService.updateObjectEntry(
+			TestPropsValues.getUserId(), objectEntry.getObjectEntryId(),
+			HashMapBuilder.<String, Serializable>put(
+				"upload", 0L
+			).build(),
+			ServiceContextTestUtil.getServiceContext());
+
+		AssertUtils.assertFailure(
+			NoSuchFileEntryException.class,
+			StringBundler.concat(
+				"No FileEntry exists with the key {fileEntryId=",
+				persistedFileEntryId1, "}"),
+			() -> _dlAppLocalService.getFileEntry(persistedFileEntryId1));
+
+		// Delete
```

Comment:
@devpedrotavares change to `// Delete object entry should delete existing files`